pkg/controller/integrationkit/build.go

```diff
@@ -133,8 +133,10 @@ func (action *buildAction) handleBuildSubmitted(ctx context.Context, kit *v1.Int
 			}
 		}
 		// The build operation, when executed as a Pod, should be executed by a container image containing the
-		// `kamel builder` command. Likely the same image running the operator should be fine.
-		buildConfig.ToolImage = platform.OperatorImage
+		// `kamel builder` command. If not specified, likely the same image running the operator should be fine.
```

Comment:
I think we must force the image to be the one on which the operator is running. Please, open a separate PR to discuss this separately.

Reply:
Done in #5110